src/packages/documents/documents/modals/rollback/rollback-modal.element.ts

```diff
@@ -0,0 +1,380 @@
+						<div>
+							<p class="rollback-item-date">
+								<umb-localize-date date="${item.date}" .options=${this.#localizeDateOptions}></umb-localize-date>
+							</p>
+							<p>${item.user}</p>
+							<p>${item.isCurrentlyPublishedVersion ? this.localize.term('rollback_currentPublishedVersion') : ''}</p>
+						</div>
+						<uui-button
+							look="secondary"
+							@click=${(event: Event) => this.#onPreventCleanup(event, item.id, !item.preventCleanup)}>
+							${item.preventCleanup
+								? this.localize.term('contentTypeEditor_historyCleanupEnableCleanup')
+								: this.localize.term('contentTypeEditor_historyCleanupPreventCleanup')}
+						</uui-button>
+					</div>
+				`;
+			},
+		)}`;
+	}
+
+	#renderCurrentVersion() {
+		if (!this.currentVersion) return;
+
+		let draftValues =
+			(this.#workspaceContext?.getData()?.values as Array<{ alias: string; culture: string; value: any }>) ?? [];
+
+		draftValues = draftValues.filter((x) => x.culture === this.currentCulture || !x.culture); // When invariant, culture is undefined or null.
+
+		const diffs = this.currentVersion.properties.map((item) => {
+			const draftValue = draftValues.find((x) => x.alias === item.alias);
+
+			if (!draftValue) return;
+
+			const draftValueString = trimQuotes(JSON.stringify(draftValue.value));
+			const versionValueString = trimQuotes(JSON.stringify(item.value));
+
+			const diff = diffWords(draftValueString, versionValueString);
+			return { alias: item.alias, diff };
+		});
+
+		function trimQuotes(str: string): string {
+			return str.replace(/^['"]|['"]$/g, '');
+		}
+
+		return html`
+			<div id="diff">
+				<p>name: ${this.currentVersion.name}</p>
+				${repeat(
+					this.currentVersion.properties,
+					(item) => item.alias,
+					(item) => {
+						const diff = diffs.find((x) => x?.alias === item.alias);
+						return html`
+							<p>
+								${item.alias}:
+								${diff
+									? diff.diff.map((part) =>
+											part.added
+												? html`<span class="added">${part.value}</span>`
+												: part.removed
+													? html`<span class="removed">${part.value}</span>`
+													: part.value,
+										)
+									: JSON.stringify(item.value)}
+							</p>
+						`;
+					},
+				)}
+			</div>
+		`;
+	}
+
+	get currentVersionHeader() {
+		return (
+			this.localize.date(this.currentVersion?.date || '', this.#localizeDateOptions) + ' - ' + this.currentVersion?.user
+		);
+	}
+
+	render() {
+		return html`
+			<umb-body-layout headline="Rollback">
+				<div id="main">
+					<uui-box headline=${this.localize.term('rollback_versions')} id="box-left">
+						<div>${this.#renderVersions()}</div>
+					</uui-box>
+					<uui-box headline=${this.currentVersionHeader} id="box-right"> ${this.#renderCurrentVersion()} </uui-box>
+				</div>
+				<umb-footer-layout slot="footer">
+					<uui-button slot="actions" look="secondary" @click=${this.#onCancel}
+						>${this.localize.term('general_cancel')}</uui-button
```

Comment:
Consider moving this to the label property instead, otherwise the button won't have a label. It will show the text automatically in the default slot if you don't have anything else there.